From 0821fe492a6377622b1774ee58e4ebae5e887e53 Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Mon, 6 May 2024 17:45:07 +0800
Subject: [PATCH 105/137] mtk: hostapd: support global_comm_band for muru
 manual config

Add support new argument global_comm_band for muru manual config
command. This argument can be used to specify the band to apply manual
config.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>
---
 hostapd/ctrl_iface.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/hostapd/ctrl_iface.c b/hostapd/ctrl_iface.c
index e534e0c63..d7d1a0159 100644
--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -4544,6 +4544,9 @@ hostapd_ctrl_iface_set_mu(struct hostapd_data *hapd, char *cmd,
 			dl->ru[i] = host_to_le16(val[i]);
 		os_free(val);
 		muru->cfg_dl |= host_to_le32(MURU_FIXED_DL_TONE_PLAN);
+	} else if (os_strcmp(config, "global_comm_band") == 0) {
+		comm->band = (u8)atoi(value);
+		muru->cfg_comm |= host_to_le32(MURU_COMM_BAND);
 	} else {
 		wpa_printf(MSG_ERROR,
 			   "Unsupported parameter %s for SET_MU", config);
-- 
2.18.0

